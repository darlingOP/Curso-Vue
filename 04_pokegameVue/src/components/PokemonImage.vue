<template>
    <div class="pokemon-container">
        <img v-if="!showPokemon" :src="imgSrc" 
         class="hidden-pokemon" 
         alt="pokemon img">
     <img v-else :src="imgSrc" 
        class="fade-in" 
        alt="pokemon img">
    </div>
    
</template>
<script>
export default {
    props:{
        pokeId:{
            type:Number,
            required: true,
        },
        showPokemon:{
            type:Boolean,
            required:true,
            default:false
        }
    },
    computed:{
        imgSrc(){
            return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokeId}.svg`
        }
    }
}
</script>
<style scoped>
   /* Pokemon Picture */
.pokemon-container {
    height: 200px;
}
img {
    height: 200px;
    user-select: none;
    -moz-user-select: none;
    -ms-user-select: none;
    -webkit-user-drag: none;
    -webkit-user-select: none;
}
.hidden-pokemon {
    filter: brightness(0);
}


</style>