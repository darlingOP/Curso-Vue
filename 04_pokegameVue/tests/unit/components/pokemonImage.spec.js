import { shallowMount } from "@vue/test-utils";
import PokemonImage from '@/components/PokemonImage'

describe('Pokemon Image Component',()=>{

    test('debe de hacer match con el snaptshot', () => {
        const wrapper = shallowMount(PokemonImage,{
            props:{
                pokeId:1,
                showPokemon:false,
            }
        })

        expect(wrapper.html()).toMatchSnapshot()
    });

    test('debe de mostrar la imagen oculta y el pokemon 100 ', () => {
        const wrapper = shallowMount(PokemonImage,{
            props:{
                pokeId:100,
                showPokemon:false
            }
        })

        //comprobar que solo se muestra la imagen oculta
        const [img1,img2] = wrapper.findAll('img')

        expect(img1.exists()).toBeTruthy()
        expect(img2).toBe(undefined)

        expect(img1.classes('hidden-pokemon')).toBe(true)
    });

    test('debe mostrar el pokemon si el showPicture es true', () => {
        const wrapper = shallowMount(PokemonImage,{
            props:{
                pokeId:100,
                showPokemon:true
            }
        })

        const img1 = wrapper.find('img')

        expect(img1.exists()).toBeTruthy()

        expect(img1.classes('fade-in')).toBe(true)

        expect(img1.attributes('src')).toBe('https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg')

    });
} )