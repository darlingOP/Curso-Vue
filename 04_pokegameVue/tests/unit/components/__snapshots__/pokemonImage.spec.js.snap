// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Image Component debe de hacer match con el snaptshot 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" class="hidden-pokemon" alt="pokemon img"></div>`;
